### PR TITLE
Httprequest parsing

### PR DESCRIPTION
**Pull request by say2prabhash**

### CODE REVIEW

File: WebServer.Core/Internal/HttpAdapter.cs [hunk 1]
```diff
@@ -8,35 +9,70 @@ public class HttpAdapter : IHttpAdapter
...
+            try
             {
-                for (int k = 0; k < s[i].Length; k++)
+                BigString clientRequestString = new BigString(payload);
+
+                byte[] b = Encoding.ASCII.GetBytes("\r\n");
+                BigString[] s = clientRequestString.Split(b);
+                b = Encoding.ASCII.GetBytes(" /");
+                BigString[] s1 = s[0].Split(b);
+                clientRequest.Verb = s1[0].ToString();
```
Comment (pkTavisca) [inline WebServer.Core/Internal/HttpAdapter.cs]: Please improve the names of variables.

Comment (say2prabhash) [inline WebServer.Core/Internal/HttpAdapter.cs]: Ok.

File: WebServer.Core/Internal/HttpAdapter.cs [hunk 2]
```diff
@@ -48,19 +48,19 @@ public HttpRequest GetRequest(byte[] payload)
...
                 {
-                    if (!(s[i].ToString()).Equals(""))
+                    if (!(parseMethodUrlString[i].ToString()).Equals(""))
                     {
-                        string tempString = s[i].ToString();
-                        string[] s2 = tempString.Split(new string[] { ":" },StringSplitOptions.None);
-                        clientRequest.Headers.Add(s2[0], s2[1]);
+                        string tempString = parseMethodUrlString[i].ToString();
+                        string[] headerParseString = tempString.Split(new string[] { ":" },StringSplitOptions.None);
+                        clientRequest.Headers.Add(headerParseString[0], headerParseString[1]);
```
Comment (pkTavisca) [inline WebServer.Core/Internal/HttpAdapter.cs]: Please fix the exception.

File: WebServer.Core/Internal/HttpAdapter.cs [hunk 3]
```diff
@@ -48,19 +48,19 @@ public HttpRequest GetRequest(byte[] payload)
...
-                b = Encoding.ASCII.GetBytes(" /");
-                BigString[] s1 = s[0].Split(b);
-                clientRequest.Verb = s1[0].ToString();
-                clientRequest.Url = s1[1].ToString();
+                byte[] splittingByte = Encoding.ASCII.GetBytes("\r\n");
+                BigString[] splittedRequest = clientRequestString.Split(splittingByte);
+                splittingByte = Encoding.ASCII.GetBytes(" /");
+                BigString[] parseMethodUrlString = splittedRequest[0].Split(splittingByte);
+                clientRequest.Verb = parseMethodUrlString[0].ToString();
+                clientRequest.Url = parseMethodUrlString[1].ToString();
```
Comment (pkTavisca) [inline WebServer.Core/Internal/HttpAdapter.cs]: Please fix the URL.

File: WebServer.Core/Internal/HttpAdapter.cs [hunk 4]
```diff
@@ -56,10 +56,10 @@ public HttpRequest GetRequest(byte[] payload)
...
                 for(int i=1;;i++)
                 {
-                    if (!(parseMethodUrlString[i].ToString()).Equals(""))
+                    if (!(splittedRequest[i].ToString()).Equals(""))
                     {
-                        string tempString = parseMethodUrlString[i].ToString();
-                        string[] headerParseString = tempString.Split(new string[] { ":" },StringSplitOptions.None);
+                        string tempString = splittedRequest[i].ToString();
+                        string[] headerParseString = tempString.Split(new string[] { ": " },StringSplitOptions.None);
                         clientRequest.Headers.Add(headerParseString[0], headerParseString[1]);
```
Comment (pkTavisca) [inline WebServer.Core/Internal/HttpAdapter.cs]: Please use this format : 
`dict[key] = value;`

**Pull request merged**